{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "### this script sorts a hmmer output based on coverage and E-value\n",
    "### Before running this script run these two commands from terminal\n",
    "# 1. build an HMM from your seed alignment\n",
    "# hmmbuild amylase.hmm seed_alignment.fasta\n",
    "\n",
    "# 2. search the Coelicolor proteome\n",
    "# hmmsearch -E 3 --domtblout amylase_hmm_domains.tbl --tblout amylase_hmm_hits.tbl --cpu 4 amylase.hmm coelicolor_proteins.faa > amylase_hmm.out\n",
    "\n",
    "### That creates a hmmer profile based on a fasta alignment file\n",
    "### then using that hmmmer profile it searces the coelicolor proteome for proteins with similar domains\n",
    "\n",
    "\n",
    "import pandas as pd\n",
    "import os\n",
    "import glob\n",
    "\n",
    "input_folder = '/Users/andreaslawaetz/Streptomyces/Angeliga_project/SCOG/amylase_inhibitors/acarviosin/gator_redo/results21_Amy_is_only_Opt_allBGC_we30/hmmer_search'\n",
    "output_folder = '/Users/andreaslawaetz/Streptomyces/Angeliga_project/SCOG/amylase_inhibitors/acarviosin/gator_redo/results21_Amy_is_only_Opt_allBGC_we30/hmmer_search_tsv'\n",
    "os.makedirs(output_folder, exist_ok = True)\n",
    "\n",
    "accessions = []\n",
    "inhibitor = []\n",
    "for file in glob.glob(input_folder + '/*.tbl'):\n",
    "    rows = []\n",
    "    with open(file) as f:\n",
    "        for line in f:\n",
    "            if line.startswith(\"#\") or not line.strip():\n",
    "                continue  # skip comments and blank lines\n",
    "\n",
    "            parts = line.strip().split()\n",
    "            # The HMMER domtblout format has 22 fixed columns before description\n",
    "            fixed_fields = parts[:18]\n",
    "            description = \" \".join(parts[18:]) if len(parts) > 18 else \"\"\n",
    "            rows.append(fixed_fields + [description])\n",
    "\n",
    "    # Convert to DataFrame\n",
    "    df = pd.DataFrame(rows, columns=[\n",
    "        \"target_name\", \"accession\", \"query_name\", \"accession\", \"E_value_1\", \"score\", \"bias\",\n",
    "        \"E_value_2\", \"score\", \"bias\", \"exp\", \"reg\", \"clu\",\n",
    "        \"ov\", \"env\", \"dom\", \"rep\", \"inc\", \"description\"\n",
    "    ])\n",
    "\n",
    "    df = df.drop(columns=[\"exp\", \"reg\", \"clu\",\n",
    "        \"ov\", \"env\", \"dom\", \"rep\", \"inc\"])\n",
    "    \n",
    "    # Convert E-value columns to numeric (coerce errors to NaN)\n",
    "    df[\"E_value_1\"] = pd.to_numeric(df[\"E_value_1\"], errors=\"coerce\")\n",
    "    df[\"E_value_2\"] = pd.to_numeric(df[\"E_value_2\"], errors=\"coerce\")\n",
    "\n",
    "    # Drop rows with missing or invalid E-values\n",
    "    df = df.dropna(subset=[\"E_value_1\"])\n",
    "\n",
    "    # Filter: only keep significant hits (E-value ≤ 1e-10)\n",
    "    df = df[df[\"E_value_1\"] <= 1e-10]\n",
    "    \n",
    "    accessions.append(os.path.basename(file)[:-9])\n",
    "    inhibitor.append(len(df))\n",
    "\n",
    "\n",
    "    # Save\n",
    "    df.to_csv(output_folder + f'/{os.path.basename(file)}.tsv', sep=\"\\t\", index=False)\n",
    "\n",
    "# master_inhibitor = pd.DataFrame()\n",
    "# master_inhibitor['tip_label'] = accessions\n",
    "# master_inhibitor['tendamistatPFAM'] = inhibitor\n",
    "\n",
    "# master_table = '/Users/andreaslawaetz/Streptomyces/Angeliga_project/SCOG/amylase_inhibitors/Master_table_with_amyPhyloGroups_known_clusters_and_inhibitors_GatorAcarviosin_GatorAcarbose_gatorValidamycinA_gatorBafilomycin_tendamisatPFAM_tendamisatSMART.csv'\n",
    "# master_df = pd.read_csv(master_table, sep = ',', header = 0)\n",
    "\n",
    "# merged_df = pd.merge(master_df, master_inhibitor, on=\"tip_label\", how=\"left\")\n",
    "# merged_df.to_csv('/Users/andreaslawaetz/Streptomyces/Angeliga_project/SCOG/amylase_inhibitors/Master_table_with_amyPhyloGroups_known_clusters_and_inhibitors_GatorAcarviosin_GatorAcarbose_gatorValidamycinA_gatorBafilomycin_tendamisatPFAM_tendamisatSMART_tendamisatPFAM.csv', sep = ',', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tip_label</th>\n",
       "      <th>2dos</th>\n",
       "      <th>amglyccycl</th>\n",
       "      <th>aminocoumarin</th>\n",
       "      <th>aminopolycarboxylic-acid</th>\n",
       "      <th>arylpolyene</th>\n",
       "      <th>atropopeptide</th>\n",
       "      <th>azole-containing-RiPP</th>\n",
       "      <th>azoxy-crosslink</th>\n",
       "      <th>azoxy-dimer</th>\n",
       "      <th>...</th>\n",
       "      <th>transAT-PKS-like</th>\n",
       "      <th>triceptide</th>\n",
       "      <th>number_amylases</th>\n",
       "      <th>GH_13_occurences</th>\n",
       "      <th>Group_1_amylase</th>\n",
       "      <th>Group_2_amylase</th>\n",
       "      <th>Group_3_amylase</th>\n",
       "      <th>resistant_amylase</th>\n",
       "      <th>acarbose_BGC</th>\n",
       "      <th>tendamistat</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCF_000009765.2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCF_000010605.1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>13.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCF_000092385.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCF_000147815.2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCF_000225525.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>15.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290</th>\n",
       "      <td>GCF_900119365.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>291</th>\n",
       "      <td>GCF_900187385.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>13.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>292</th>\n",
       "      <td>GCF_900206255.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>293</th>\n",
       "      <td>GCF_900215595.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>15.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294</th>\n",
       "      <td>GCF_900215615.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>17.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>295 rows × 77 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           tip_label  2dos  amglyccycl  aminocoumarin  \\\n",
       "0    GCF_000009765.2     0           0              0   \n",
       "1    GCF_000010605.1     0           1              0   \n",
       "2    GCF_000092385.1     0           0              0   \n",
       "3    GCF_000147815.2     0           0              0   \n",
       "4    GCF_000225525.1     0           0              0   \n",
       "..               ...   ...         ...            ...   \n",
       "290  GCF_900119365.1     0           0              0   \n",
       "291  GCF_900187385.1     0           0              0   \n",
       "292  GCF_900206255.1     0           0              0   \n",
       "293  GCF_900215595.1     0           0              0   \n",
       "294  GCF_900215615.1     0           0              0   \n",
       "\n",
       "     aminopolycarboxylic-acid  arylpolyene  atropopeptide  \\\n",
       "0                           1            0              0   \n",
       "1                           0            0              0   \n",
       "2                           0            1              0   \n",
       "3                           0            0              0   \n",
       "4                           0            0              0   \n",
       "..                        ...          ...            ...   \n",
       "290                         0            0              0   \n",
       "291                         0            0              1   \n",
       "292                         0            0              0   \n",
       "293                         1            0              0   \n",
       "294                         0            0              0   \n",
       "\n",
       "     azole-containing-RiPP  azoxy-crosslink  azoxy-dimer  ...  \\\n",
       "0                        0                0            0  ...   \n",
       "1                        0                0            0  ...   \n",
       "2                        1                0            0  ...   \n",
       "3                        0                0            0  ...   \n",
       "4                        0                0            0  ...   \n",
       "..                     ...              ...          ...  ...   \n",
       "290                      0                0            0  ...   \n",
       "291                      0                0            0  ...   \n",
       "292                      0                1            0  ...   \n",
       "293                      0                0            0  ...   \n",
       "294                      1                0            0  ...   \n",
       "\n",
       "     transAT-PKS-like  triceptide  number_amylases  GH_13_occurences  \\\n",
       "0                   0           1                2              16.0   \n",
       "1                   0           0                3              13.0   \n",
       "2                   0           0                0              10.0   \n",
       "3                   0           1                1              11.0   \n",
       "4                   0           0                2              15.0   \n",
       "..                ...         ...              ...               ...   \n",
       "290                 0           0                1              11.0   \n",
       "291                 0           1                2              13.0   \n",
       "292                 0           0                3              18.0   \n",
       "293                 0           0                2              15.0   \n",
       "294                 0           0                2              17.0   \n",
       "\n",
       "     Group_1_amylase  Group_2_amylase  Group_3_amylase  resistant_amylase  \\\n",
       "0                1.0              1.0              0.0                  0   \n",
       "1                1.0              1.0              1.0                  0   \n",
       "2                0.0              0.0              0.0                  0   \n",
       "3                1.0              0.0              0.0                  0   \n",
       "4                1.0              1.0              0.0                  1   \n",
       "..               ...              ...              ...                ...   \n",
       "290              1.0              0.0              0.0                  0   \n",
       "291              1.0              1.0              0.0                  0   \n",
       "292              1.0              2.0              0.0                  0   \n",
       "293              1.0              1.0              0.0                  0   \n",
       "294              1.0              1.0              0.0                  0   \n",
       "\n",
       "     acarbose_BGC  tendamistat  \n",
       "0               0            1  \n",
       "1               0            0  \n",
       "2               0            0  \n",
       "3               0            0  \n",
       "4               0            0  \n",
       "..            ...          ...  \n",
       "290             0            0  \n",
       "291             0            0  \n",
       "292             0            0  \n",
       "293             0            0  \n",
       "294             0            0  \n",
       "\n",
       "[295 rows x 77 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tip_label</th>\n",
       "      <th>2dos</th>\n",
       "      <th>amglyccycl</th>\n",
       "      <th>aminocoumarin</th>\n",
       "      <th>aminopolycarboxylic-acid</th>\n",
       "      <th>arylpolyene</th>\n",
       "      <th>atropopeptide</th>\n",
       "      <th>azole-containing-RiPP</th>\n",
       "      <th>azoxy-crosslink</th>\n",
       "      <th>azoxy-dimer</th>\n",
       "      <th>...</th>\n",
       "      <th>thioamide-NRP</th>\n",
       "      <th>thioamitides</th>\n",
       "      <th>transAT-PKS</th>\n",
       "      <th>transAT-PKS-like</th>\n",
       "      <th>triceptide</th>\n",
       "      <th>number_amylases</th>\n",
       "      <th>GH_13_occurences</th>\n",
       "      <th>Group_1_amylase</th>\n",
       "      <th>Group_2_amylase</th>\n",
       "      <th>Group_3_amylase</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCF_000009765.2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCF_000010605.1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>13.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCF_000092385.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCF_000147815.2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCF_000225525.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>15.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290</th>\n",
       "      <td>GCF_900119365.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>11.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>291</th>\n",
       "      <td>GCF_900187385.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>13.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>292</th>\n",
       "      <td>GCF_900206255.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>18.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>293</th>\n",
       "      <td>GCF_900215595.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>15.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294</th>\n",
       "      <td>GCF_900215615.1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>17.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>295 rows × 74 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           tip_label  2dos  amglyccycl  aminocoumarin  \\\n",
       "0    GCF_000009765.2     0           0              0   \n",
       "1    GCF_000010605.1     0           1              0   \n",
       "2    GCF_000092385.1     0           0              0   \n",
       "3    GCF_000147815.2     0           0              0   \n",
       "4    GCF_000225525.1     0           0              0   \n",
       "..               ...   ...         ...            ...   \n",
       "290  GCF_900119365.1     0           0              0   \n",
       "291  GCF_900187385.1     0           0              0   \n",
       "292  GCF_900206255.1     0           0              0   \n",
       "293  GCF_900215595.1     0           0              0   \n",
       "294  GCF_900215615.1     0           0              0   \n",
       "\n",
       "     aminopolycarboxylic-acid  arylpolyene  atropopeptide  \\\n",
       "0                           1            0              0   \n",
       "1                           0            0              0   \n",
       "2                           0            1              0   \n",
       "3                           0            0              0   \n",
       "4                           0            0              0   \n",
       "..                        ...          ...            ...   \n",
       "290                         0            0              0   \n",
       "291                         0            0              1   \n",
       "292                         0            0              0   \n",
       "293                         1            0              0   \n",
       "294                         0            0              0   \n",
       "\n",
       "     azole-containing-RiPP  azoxy-crosslink  azoxy-dimer  ...  thioamide-NRP  \\\n",
       "0                        0                0            0  ...              0   \n",
       "1                        0                0            0  ...              0   \n",
       "2                        1                0            0  ...              0   \n",
       "3                        0                0            0  ...              0   \n",
       "4                        0                0            0  ...              0   \n",
       "..                     ...              ...          ...  ...            ...   \n",
       "290                      0                0            0  ...              0   \n",
       "291                      0                0            0  ...              0   \n",
       "292                      0                1            0  ...              0   \n",
       "293                      0                0            0  ...              0   \n",
       "294                      1                0            0  ...              0   \n",
       "\n",
       "     thioamitides  transAT-PKS  transAT-PKS-like  triceptide  number_amylases  \\\n",
       "0               0            0                 0           1                2   \n",
       "1               0            0                 0           0                3   \n",
       "2               1            1                 0           0                0   \n",
       "3               0            0                 0           1                1   \n",
       "4               1            0                 0           0                2   \n",
       "..            ...          ...               ...         ...              ...   \n",
       "290             0            0                 0           0                1   \n",
       "291             1            0                 0           1                2   \n",
       "292             1            0                 0           0                3   \n",
       "293             0            0                 0           0                2   \n",
       "294             0            0                 0           0                2   \n",
       "\n",
       "     GH_13_occurences  Group_1_amylase  Group_2_amylase  Group_3_amylase  \n",
       "0                16.0              1.0              1.0              0.0  \n",
       "1                13.0              1.0              1.0              1.0  \n",
       "2                10.0              0.0              0.0              0.0  \n",
       "3                11.0              1.0              0.0              0.0  \n",
       "4                15.0              1.0              1.0              0.0  \n",
       "..                ...              ...              ...              ...  \n",
       "290              11.0              1.0              0.0              0.0  \n",
       "291              13.0              1.0              1.0              0.0  \n",
       "292              18.0              1.0              2.0              0.0  \n",
       "293              15.0              1.0              1.0              0.0  \n",
       "294              17.0              1.0              1.0              0.0  \n",
       "\n",
       "[295 rows x 74 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "master_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tip_label</th>\n",
       "      <th>tendamistat</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCF_016741855.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCF_000931445.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCF_004125245.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCF_008704515.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCF_017876235.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GCF_000009765.2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GCF_000739045.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GCF_017526105.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GCF_004028635.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GCF_000700005.2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GCF_000720675.1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GCF_000719865.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GCF_000718375.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>GCF_001509795.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>GCF_002188335.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GCF_008704575.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>GCF_003947265.2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GCF_014649995.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GCF_011008945.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GCF_001746425.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>GCF_000968685.2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GCF_002251155.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>GCF_002005565.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GCF_009377175.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>GCF_008973465.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>GCF_018070175.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>GCF_000716625.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>GCF_001509775.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GCF_003945525.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>GCF_014649515.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>GCF_001514235.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GCF_003143935.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>GCF_014203595.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>GCF_009852415.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>GCF_014203535.1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>GCF_002156055.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>GCF_014205315.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>GCF_001905855.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>GCF_900105245.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>GCF_000718455.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>GCF_002156095.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>GCF_002802945.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>GCF_003363195.1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          tip_label  tendamistat\n",
       "0   GCF_016741855.1            1\n",
       "1   GCF_000931445.1            1\n",
       "2   GCF_004125245.1            1\n",
       "3   GCF_008704515.1            1\n",
       "4   GCF_017876235.1            1\n",
       "5   GCF_000009765.2            1\n",
       "6   GCF_000739045.1            1\n",
       "7   GCF_017526105.1            1\n",
       "8   GCF_004028635.1            1\n",
       "9   GCF_000700005.2            1\n",
       "10  GCF_000720675.1            2\n",
       "11  GCF_000719865.1            1\n",
       "12  GCF_000718375.1            1\n",
       "13  GCF_001509795.1            1\n",
       "14  GCF_002188335.1            1\n",
       "15  GCF_008704575.1            1\n",
       "16  GCF_003947265.2            1\n",
       "17  GCF_014649995.1            1\n",
       "18  GCF_011008945.1            1\n",
       "19  GCF_001746425.1            1\n",
       "20  GCF_000968685.2            1\n",
       "21  GCF_002251155.1            1\n",
       "22  GCF_002005565.1            1\n",
       "23  GCF_009377175.1            1\n",
       "24  GCF_008973465.1            1\n",
       "25  GCF_018070175.1            1\n",
       "26  GCF_000716625.1            1\n",
       "27  GCF_001509775.1            1\n",
       "28  GCF_003945525.1            1\n",
       "29  GCF_014649515.1            1\n",
       "30  GCF_001514235.1            1\n",
       "31  GCF_003143935.1            1\n",
       "32  GCF_014203595.1            1\n",
       "33  GCF_009852415.1            1\n",
       "34  GCF_014203535.1            2\n",
       "35  GCF_002156055.1            1\n",
       "36  GCF_014205315.1            1\n",
       "37  GCF_001905855.1            1\n",
       "38  GCF_900105245.1            1\n",
       "39  GCF_000718455.1            1\n",
       "40  GCF_002156095.1            1\n",
       "41  GCF_002802945.1            1\n",
       "42  GCF_003363195.1            1"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "master_inhibitor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pandas",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
